### PR TITLE
Remove the use of APIIdentifier and getAPI by identifier part 1

### PR DESCRIPTION
**Pull request by tharikaGitHub**
This PR does the following changes.

1. Remove the use of APIIdentifier and replace with UUID for method apimgtDAO.getAPIID(Identifier)
2. Replace the use of method apiProvider.getAPI(Identifier) with apiProvider.getAPIbyUUID(Identifier,organization) in publisher.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java [hunk 1]
```diff
@@ -105,13 +105,13 @@ public void addComment(APIIdentifier identifier, String s, String user) throws A
...
-        return super.addComment(identifier, comment, user);
+    public String addComment(String uuid, Comment comment, String user) throws APIManagementException {
+        return super.addComment(uuid, comment, user);
     }
 
     @Override
-    public void deleteComment(APIIdentifier identifier, String commentId) throws APIManagementException {
-        super.deleteComment(identifier,commentId);
+    public void deleteComment(String uuid, String commentId) throws APIManagementException {
+        super.deleteComment(uuid,commentId);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java:114]: Format

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 2]
```diff
@@ -1315,6 +1315,8 @@
 
     public static final String GET_API_ID_SQL =
             "SELECT API.API_ID FROM AM_API API WHERE API.API_PROVIDER = ? AND API.API_NAME = ? AND API.API_VERSION = ? ";
+    public static final String GET_API_ID_SQL_NEW =
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:1318]: lets rename this to GET_API_ID_SQL_BY_UUID or use GET_API_ID_SQL

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 3]
```diff
@@ -1651,6 +1653,9 @@
     public static final String REMOVE_FROM_API_SQL =
             "DELETE FROM AM_API WHERE API_PROVIDER=? AND API_NAME=? AND API_VERSION=? ";
 
+    public static final String REMOVE_FROM_API_SQL_NEW =
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:1656]: same here.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java [hunk 4]
```diff
@@ -142,14 +142,7 @@
...
 import java.lang.reflect.Modifier;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.SortedMap;
-import java.util.TreeMap;
+import java.util.*;
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java:145]: let's not use wildcard imports

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java (hunk 1 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10487/commits/f7cce7cd884bf9fc58efecaaad4359ffe282bc84

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java (hunk 2 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10487/commits/f7cce7cd884bf9fc58efecaaad4359ffe282bc84

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java (hunk 3 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10487/commits/f7cce7cd884bf9fc58efecaaad4359ffe282bc84

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 4 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10487/commits/f7cce7cd884bf9fc58efecaaad4359ffe282bc84

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 5]
```diff
@@ -100,31 +100,7 @@
...
-import org.wso2.carbon.apimgt.api.model.KeyManagerConfiguration;
-import org.wso2.carbon.apimgt.api.model.KeyManagerConnectorConfiguration;
-import org.wso2.carbon.apimgt.api.model.Provider;
-import org.wso2.carbon.apimgt.api.model.ResourceFile;
-import org.wso2.carbon.apimgt.api.model.Scope;
-import org.wso2.carbon.apimgt.api.model.Tier;
-import org.wso2.carbon.apimgt.api.model.URITemplate;
-import org.wso2.carbon.apimgt.api.model.VHost;
-import org.wso2.carbon.apimgt.api.model.WebsubSubscriptionConfiguration;
+import org.wso2.carbon.apimgt.api.model.*;
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:103]: Remove wild card import

Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10487/commits/ed9f558f4577d0c835debc234a1e657e1c4422e3

**Pull request merged**